### PR TITLE
Cr dev

### LINKED ISSUE #21: Default conversion _id to id
>  // Why couldn't Mongo just use "id"?

No shit?! 

@jmdobry How do you feel about making default functionality to convert `id` to `_id` and vica versa with adapter interactions?

We can still have the config `idAttribute` property to be specified for override.

### PR DESCRIPTION
**Pull request by crobinson42**
Fixes #21 

> Note: This project uses `semantic-release` to auto publish npm packages on successful builds.
> The commit messages must follow AngularJS Commit Msg Conventions: https://docs.google.com/document/d/1QrDFcIiPjSLDn3EL15IJygNPiHORgU1_OOAqWjiDU5Y/edit#heading=h.uyo6cb12dt6w
- [x] - My commit messages follow AngularJS Commit Msg Conventions
- [ ] - If applicable, appropriate JSDoc comments were updated in source code
- [ ] - If applicable, appropriate changes to js-data.io docs have been suggested ("Suggest Edits" button)
- [x] - Unit test coverage did not decrease & appropriate tests changes made

### CODE REVIEW

**Comment by jmdobry** [comment]
Why does it say merging will decrease coverage by 15%?

**Comment by crobinson42** [comment]
I'm unsure about codecov reports. When I started on the branch coverage was
in the 70% range. Is it reporting coverage diff from a release?

**Comment by crobinson42** [comment]
@jmdobry Any thoughts, ideas, or recommendations on this PR?

Also, any idea on the hanging codecov status? I'm sure the code coverage isn't decreasing.. am I missing something here?

File: src/index.js [hunk 1]
```diff
@@ -26,6 +26,15 @@ const DEFAULTS = {
   translateObjectIDs: false,
 
   /**
+   * Convert the MongoDB _id to 'id'
+   *
+   * @name MongoDBAdapter#convertMongo_id
+   * @type {Boolean}
+   * @default false
+   */
+  convertMongo_id: false,
```
Comment (jmdobry) [inline src/index.js:35]: Can we change this option to be fully camelCase?

File: src/index.js [hunk 2]
```diff
@@ -26,6 +26,15 @@ const DEFAULTS = {
   translateObjectIDs: false,
 
   /**
+   * Convert the MongoDB _id to 'id'
+   *
+   * @name MongoDBAdapter#convertMongo_id
+   * @type {Boolean}
```
Comment (jmdobry) [inline src/index.js:32]: `{Boolean}` => `{boolean}`

File: src/index.js [hunk 3]
```diff
@@ -87,6 +96,7 @@ const REMOVE_OPTS_DEFAULTS = {}
 export function MongoDBAdapter (opts) {
   utils.classCallCheck(this, MongoDBAdapter)
   opts || (opts = {})
+  /* istanbul ignore if */
```
Comment (jmdobry) [inline src/index.js:99]: Why are we ignoring this?

File: src/index.js [hunk 4]
```diff
@@ -199,11 +210,12 @@ Adapter.extend({
   constructor: MongoDBAdapter,
 
   _translateObjectIDs (r, opts) {
+    if (!r) return r
```
Comment (jmdobry) [inline src/index.js:213]: `if (!r) return r`

=>

``` js
if (!r) {
  return r
}
```

File: src/index.js [hunk 5]
```diff
@@ -214,21 +226,23 @@ Adapter.extend({
...
-      r._id = typeof __id === 'string' ? __id : r._id
+      const __id = r[idKey] ? r[idKey].toString() : r[idKey]
+      r[idKey] = typeof __id === 'string' ? __id : r[idKey]
     }
     return r
   },
 
   _translateFieldObjectIDs (r) {
     const _checkFields = (r) => {
+      if (typeof r !== 'object') return
```
Comment (jmdobry) [inline src/index.js:245]: `if (typeof r !== 'object') return`

=>

``` js
if (!utils.isObject(r)) {
  return
}
```

File: src/index.js [hunk 6]
```diff
@@ -246,6 +260,34 @@ Adapter.extend({
   },
 
   /**
+   * Convert mongodb _id to id
+   *
+   * @method MongoDBAdapter#_convertMongo_id
+   * @return {*} converts the params _id key(s) to id
+   */
+  _convertMongo_id (d, opts) {
```
Comment (jmdobry) [inline src/index.js:268]: Can we make this method name fully camelCase?

File: src/index.js [hunk 7]
```diff
@@ -246,6 +260,34 @@ Adapter.extend({
...
+    if (!d) return
```
Comment (jmdobry) [inline src/index.js:269]: `if (!d) return` =>

``` js
if (!d) {
  return
}
```

File: src/index.js (hunk 6 above)
Comment (jmdobry) [inline src/index.js:268]: Can we use a more descriptive variable name than `d`?

File: src/index.js [hunk 8]
```diff
@@ -214,21 +226,23 @@ Adapter.extend({
...
     } else if (utils.isObject(r)) {
-      const __id = r._id ? r._id.toString() : r._id
-      r._id = typeof __id === 'string' ? __id : r._id
+      const __id = r[idKey] ? r[idKey].toString() : r[idKey]
+      r[idKey] = typeof __id === 'string' ? __id : r[idKey]
     }
     return r
   },
 
   _translateFieldObjectIDs (r) {
```
Comment (jmdobry) [inline src/index.js:243]: Can we use a more descriptive variable name than `r`?

File: src/index.js [hunk 9]
```diff
@@ -214,21 +226,23 @@ Adapter.extend({
    * @method MongoDBAdapter#_translateId
    * @return {*}
    */
-  _translateId (r) {
+  _translateId (r, opts) {
```
Comment (jmdobry) [inline src/index.js:229]: Can we use a more descriptive variable name than `r`?

File: src/index.js [hunk 10]
```diff
@@ -199,11 +210,12 @@ Adapter.extend({
   constructor: MongoDBAdapter,
 
   _translateObjectIDs (r, opts) {
```
Comment (jmdobry) [inline src/index.js:212]: Can we use a more descriptive variable name than `r`?

File: src/index.js [hunk 11]
```diff
@@ -246,6 +260,34 @@ Adapter.extend({
...
+   *
+   * @method MongoDBAdapter#_convertMongo_id
+   * @return {*} converts the params _id key(s) to id
+   */
+  _convertMongo_id (d, opts) {
+    if (!d) return
+    opts || (opts = {})
+    if (!this.getOpt('convertMongo_id', opts)) return d
+
+    const convert = (x) => {
```
Comment (jmdobry) [inline src/index.js:273]: Can we pull this out into a top-level standalone function? It doesn't look like it needs to create a closure over anything.

**Review by jmdobry — CHANGES_REQUESTED** [review]

**Pull request closed without merging**